pkg/service/http_service.go

```diff
@@ -19,153 +24,84 @@ type HTTPServiceConfiguration struct {
+		fmt.Sprintf("localhost:%d", s.HTTPServiceConfiguration.Port),
+		[]grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())},
+	)
 	if err != nil {
-		handleParseError(err, w)
-		return
+		log.Fatal(err)
 	}
 
-	result, reason, err := s.eval.ResolveBooleanValue(flagKey, params.DefaultValue, contextObj)
-	if err != nil {
-		handleEvaluationError(err, reason, w)
-		return
+	server := http.Server{
+		Handler: mux,
 	}
-	_ = json.NewEncoder(w).Encode(gen.ResolutionDetailsBoolean{
-		Value:  result,
-		Reason: &reason,
-	})
-}
 
-func (s Server) ResolveString(
-	w http.ResponseWriter,
-	r *http.Request,
-	flagKey gen.FlagKey,
-	params gen.ResolveStringParams,
-) {
-	var contextObj gen.Context
-	err := json.NewDecoder(r.Body).Decode(&contextObj)
+	l, err := net.Listen("tcp", fmt.Sprintf(":%d", s.HTTPServiceConfiguration.Port))
 	if err != nil {
-		handleParseError(err, w)
-		return
+		log.Fatal(err)
 	}
+	m := cmux.New(l)
 
-	result, reason, err := s.eval.ResolveStringValue(flagKey, params.DefaultValue, contextObj)
-	if err != nil {
-		handleEvaluationError(err, reason, w)
-		return
-	}
-	_ = json.NewEncoder(w).Encode(gen.ResolutionDetailsString{
-		Value:  result,
-		Reason: &reason,
-	})
-}
+	httpL := m.Match(cmux.HTTP1Fast())
+	grpcL := m.Match(cmux.HTTP2())
 
-func (s Server) ResolveNumber(
-	w http.ResponseWriter,
-	r *http.Request,
-	flagKey gen.FlagKey,
-	params gen.ResolveNumberParams,
-) {
-	var contextObj gen.Context
-	err := json.NewDecoder(r.Body).Decode(&contextObj)
-	if err != nil {
-		handleParseError(err, w)
-		return
-	}
+	go func() { handleServiceError(server.Serve(httpL)) }()
+	go func() { handleServiceError(grpcServer.Serve(grpcL)) }()
+	go func() { handleServiceError(m.Serve()) }()
 
-	result, reason, err := s.eval.ResolveNumberValue(flagKey, params.DefaultValue, contextObj)
-	if err != nil {
-		handleEvaluationError(err, reason, w)
-		return
-	}
-	_ = json.NewEncoder(w).Encode(gen.ResolutionDetailsNumber{
-		Value:  result,
-		Reason: &reason,
-	})
+	<-ctx.Done()
+	return nil
 }
 
-func (s Server) ResolveObject(
+func (s HTTPService) HTTPErrorHandler(
+	ctx context.Context,
+	m *runtime.ServeMux,
+	ma runtime.Marshaler,
 	w http.ResponseWriter,
 	r *http.Request,
-	flagKey gen.FlagKey,
-	params gen.ResolveObjectParams,
+	err error,
 ) {
-	var contextObj gen.Context
-	err := json.NewDecoder(r.Body).Decode(&contextObj)
-	if err != nil {
-		handleParseError(err, w)
-		return
+	st := status.Convert(err)
+	switch {
+	case st.Code() == codes.Unknown:
+		w.WriteHeader(http.StatusInternalServerError)
+	case st.Code() == codes.InvalidArgument:
+		w.WriteHeader(http.StatusBadRequest)
+	case st.Code() == codes.NotFound:
+		w.WriteHeader(http.StatusNotFound)
+	default:
+		w.WriteHeader(http.StatusInternalServerError)
 	}
-
-	result, reason, err := s.eval.ResolveObjectValue(flagKey, params.DefaultValue.AdditionalProperties, contextObj)
-	if err != nil {
-		handleEvaluationError(err, reason, w)
+	details := st.Details()
```

Comment:
details[0] is an `ErrorResponse`, which doesn't have a `reason`... The end result of this is that `"reason": "ERROR"` isn't returned in the HTTP response, which is probably fine. Similar to the `default-value` and the `flag-key`, the provider using the http client will know this is an error and can set the field accordingly

For reference: https://github.com/open-feature/spec/blob/main/specification/flag-evaluation.md#requirement-148

Reply:
will add, schema update PR found here
https://github.com/open-feature/schemas/pull/11